### DISCUSSION TITLE
How about adding a Kotlin template?

### DISCUSSION THREAD

**Discussion by nilwurtz** [getgauge/gauge#2325] [Ideas]
Hi everyone. 
I often use gauge in my kotlin-maven projects, and I think that there is no problem with IntelliJ plugin support.

However, `gauge init` command does not have a template for kotlin,  so I always use a template like [this](https://github.com/nilwurtz/gauge-kotlin-maven).

How about adding a template for kotlin to `gauge init` command to make it more friendly to developers who want to develop in Kotlin?

I understand that these discussions are taking place.
https://github.com/getgauge/gauge/discussions/1861

**Comment by zabil** [comment]
You can add custom templates and point to the location of a custom template for example.

```
Change template configurations.

Usage:
  gauge template [flags] [args]

Examples:
  gauge template custom-template https://example.org/templates/custom-template.zip
```

This can point to a GitHub release too. If you have a sample Kotlin file maybe you can share that link here so that others can try out your plugin as

```
gauge template kotlin https://example.org/templates/kotlin-template.zip
```

**Reply by nilwurtz** [reply]
Thank you.

Sorry for not knowing about the `gauge template` command.
I would like to create a kotlin-maven plugin release and share it here.

By the way, do you have any plans to add Gauge's default template like getgauge/template-kotlin?

**Reply by zabil** [reply]
> By the way, do you have any plans to add Gauge's default template like getgauge/template-kotlin?
 Write
 
 @nilwurtz there are no plans for this. Gauge does not have a dedicated core team and it's better for these plugins and templates to be maintained independently. Google is good at indexing gauge plugins. If you are planning to add a repo with a README on how to use it please drop a link to it.

**Reply by nilwurtz** [reply]
@zabil 
Thank you for answering.

Now then I've created [template repository](https://github.com/nilwurtz/gauge-kotlin-maven) and the release, so we can create Kotlin project with using the command below!

- Add custom template
```
gauge template kotlin_maven https://github.com/nilwurtz/gauge-kotlin-maven/releases/latest/download/kotlin_maven.zip
```

- Init with the template
```
gauge init kotlin_maven
```

I hope everyone will use it.

**Comment by nilwurtz** [comment]
Now that we can init kotlin project by using a custom template, I'm going to close this discussion.

- Add custom template
```
gauge template kotlin_maven https://github.com/nilwurtz/gauge-kotlin-maven/releases/latest/download/kotlin_maven.zip
```
- Init with the template
```
gauge init kotlin_maven
```

Thanks!
